core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts

```diff
@@ -85,21 +83,27 @@ export class EmaContentletToolsComponent implements OnInit, OnChanges {
 
     vtlFiles: MenuItem[] = [];
 
-    ACTIONS_CONTAINER_WIDTH: number; // Now is dynamic based on the page type (Headless - VTL)
+    ACTIONS_CONTAINER_WIDTH = 128; // Now is dynamic based on the page type (Headless - VTL)
 
-    ngOnInit() {
-        this.setVtlFiles();
-        this.ACTIONS_CONTAINER_WIDTH = this.contentletArea.payload.vtlFiles ? 178 : 128;
-    }
+    protected styles: Record<string, { [klass: string]: unknown }> = {};
 
     ngOnChanges(changes: SimpleChanges): void {
         if (!changes.contentletArea) {
             return;
         }
 
-        if (this.hasContentletOrContainerChanged(changes.contentletArea)) {
-            this.hideMenus();
-        }
+        this.hideMenus(); // We need to hide the menu if the contentlet changes
+        this.setVtlFiles(); // Set the VTL files for the component
+
+        this.ACTIONS_CONTAINER_WIDTH = this.contentletArea.payload.vtlFiles ? 178 : 128; // Set the width of the actions container
+
+        // If the contentlet changed, we need to update the styles
+        this.styles = {
+            bounds: this.getBoundsPosition(),
```

Comment:
This is not the same as using the method direct in the template?

Reply:
No, in this `ngOnChanges` im sure that we don't trigger this functions unless we change the `contentletArea` input